src/get-options-overrides.ts

```diff
@@ -16,13 +16,26 @@ export function getOptionsOverrides({ useTsconfigDeclarationDir, cacheRoot }: IO
 		noEmitOnError: false,
 		inlineSourceMap: false,
 		outDir: normalize(`${cacheRoot}/placeholder`), // need an outdir that is different from source or tsconfig parsing trips up. https://github.com/Microsoft/TypeScript/issues/24715
-		moduleResolution: tsModule.ModuleResolutionKind.NodeJs,
 		allowNonTsExtensions: true,
+		moduleResolution: tsModule.ModuleResolutionKind.Node10,
 	};
 
 	if (!preParsedTsconfig)
 		return overrides;
 
+	switch (preParsedTsconfig.options.moduleResolution)
+	{
+		case tsModule.ModuleResolutionKind.Node10:
+		case tsModule.ModuleResolutionKind.Node16:
+		case tsModule.ModuleResolutionKind.NodeNext:
+			overrides.moduleResolution = preParsedTsconfig.options.moduleResolution;
+			break;
+		case tsModule.ModuleResolutionKind.Classic:
+		case tsModule.ModuleResolutionKind.Bundler:
+		default:
+			overrides.moduleResolution = tsModule.ModuleResolutionKind.Node10;
```

Comment:
I'm gonna need to review this in a lot more detail (need to catch up on TS changes since 5.x hit GA and refresh myself on this codebase), but I mentioned `mjs` support and `bundler` in https://github.com/ezolenko/rollup-plugin-typescript2/issues/434#issuecomment-1407796721.

[`bundler`](https://devblogs.microsoft.com/typescript/announcing-typescript-5-0-beta/#moduleresolution-bundler) actually seemed the most appropriate at the time (since Rollup is a bundler) for `Node16`+, particularly due to the file extension issues. That's actually why I didn't implement support for `mjs` / `cjs` earlier because the file extension requirement could break many things (that and no user requested it)

Reply:
Yeah, this basically lowest effort to let users use moduleResolution without figuring out exactly what the differences are.

Reply:
So I think `bundler` should not be overwritten either. With that change and README modifications about this, I think we could approve and merge this in.

Without tests though, we won't be able to confirm some of the behavior of `node16` / `nodeNext` -- it is possible we may need to override those two to `bundler` as well